### ISSUE TITLE
no rule to make target '../scripts/postinstall.sh'

### ISSUE THREAD

**Issue by nordbit** [mapnik/node-mapnik#771]
node v. 6.10.2
gcc version 6.3.0
```  SOLINK_MODULE(target) /home/osm/nodetile/node_modules/mapnik/lib/binding/mapnik.node
  COPY Release/mapnik.node
make: *** No rule to make target '../scripts/postinstall.sh', needed by 'lib/binding/mapnik'.  Stop.
```

**Comment by springmeyer** [comment]
Can you provide the entire terminal output for the install? By default we provide binaries and I need the terminal output to figure out why they failed to install.

**Comment by 1ec5** [comment]
node v8.1.4

<details>
<summary>node-mapnik installation failure</summary>

```
> mapnik@3.6.0 install /path/to/node_modules/mapnik
> node-pre-gyp install --fallback-to-build

node-pre-gyp info it worked if it ends with ok
node-pre-gyp verb cli [ '/usr/local/Cellar/node/8.1.4/bin/node',
node-pre-gyp verb cli   '/path/to/node_modules/mapnik/node_modules/.bin/node-pre-gyp',
node-pre-gyp verb cli   'install',
node-pre-gyp verb cli   '--fallback-to-build' ]
node-pre-gyp info using node-pre-gyp@0.6.34
node-pre-gyp info using node@8.1.4 | darwin | x64
node-pre-gyp verb command install []
node-pre-gyp info check checked for "/path/to/node_modules/mapnik/lib/binding/mapnik.node" (not found)
node-pre-gyp http GET https://mapbox-node-binary.s3.amazonaws.com/mapnik/v3.6.0/node-v57-darwin-x64-Release.tar.gz
node-pre-gyp http 403 https://mapbox-node-binary.s3.amazonaws.com/mapnik/v3.6.0/node-v57-darwin-x64-Release.tar.gz
node-pre-gyp ERR! Tried to download(403): https://mapbox-node-binary.s3.amazonaws.com/mapnik/v3.6.0/node-v57-darwin-x64-Release.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for mapnik@3.6.0 and node@8.1.4 (node-v57 ABI) (falling back to source compile with node-gyp) 
node-pre-gyp http 403 status code downloading tarball https://mapbox-node-binary.s3.amazonaws.com/mapnik/v3.6.0/node-v57-darwin-x64-Release.tar.gz 
node-pre-gyp verb command build [ 'rebuild' ]
  ACTION binding_gyp_make_vector_tile_target_generate_setting Release/obj/gen/mapnik_settings.js
sh: ./mason_packages/.link/bin/mapnik-config: No such file or directory
sh: ./mason_packages/.link/bin/mapnik-config: No such file or directory
  COPY /path/to/node_modules/mapnik/lib/binding/mapnik_settings.js
  TOUCH Release/obj.target/make_vector_tile.stamp
  CXX(target) Release/obj.target/mapnik/src/mapnik_logger.o
  CXX(target) Release/obj.target/mapnik/src/node_mapnik.o
  CXX(target) Release/obj.target/mapnik/src/blend.o
  CXX(target) Release/obj.target/mapnik/src/mapnik_map.o
../src/mapnik_map.cpp:1459:69: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    info.GetReturnValue().Set(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                                    ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_color.o
../src/mapnik_color.cpp:186:63: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                              ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_geometry.o
../src/mapnik_geometry.cpp:91:63: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                              ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_feature.o
../src/mapnik_feature.cpp:117:73: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        info.GetReturnValue().Set(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                                        ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_feature.cpp:135:63: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                              ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
2 warnings generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_image.o
../src/mapnik_image.cpp:1618:81: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        v8::Local<v8::Object> image_obj = Nan::New(constructor)->GetFunction()->NewInstance(1, &ext);
                                                                                ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:1736:67: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                                  ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:2080:81: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        v8::Local<v8::Object> image_obj = Nan::New(constructor)->GetFunction()->NewInstance(1, &ext);
                                                                                ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:2266:67: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                                  ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:2375:75: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
                return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                                          ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:2511:81: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        v8::Local<v8::Object> image_obj = Nan::New(constructor)->GetFunction()->NewInstance(1, &ext);
                                                                                ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:2721:67: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                                  ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:2958:81: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        v8::Local<v8::Object> image_obj = Nan::New(constructor)->GetFunction()->NewInstance(1, &ext);
                                                                                ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:3173:81: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        v8::Local<v8::Object> image_obj = Nan::New(constructor)->GetFunction()->NewInstance(1, &ext);
                                                                                ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:3291:85: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        v8::Local<v8::Value> image_instance = Nan::New(constructor)->GetFunction()->NewInstance(1, &ext);
                                                                                    ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:3346:71: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
            return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                                      ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_image.cpp:3515:81: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        v8::Local<v8::Object> image_obj = Nan::New(constructor)->GetFunction()->NewInstance(1, &ext);
                                                                                ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
12 warnings generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_image_view.o
../src/mapnik_image_view.cpp:102:63: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                              ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_grid.o
  CXX(target) Release/obj.target/mapnik/src/mapnik_grid_view.o
../src/mapnik_grid_view.cpp:86:63: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                              ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_memory_datasource.o
../src/mapnik_memory_datasource.cpp:115:64: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape( Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                               ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_palette.o
  CXX(target) Release/obj.target/mapnik/src/mapnik_projection.o
  CXX(target) Release/obj.target/mapnik/src/mapnik_layer.o
../src/mapnik_layer.cpp:114:63: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                              ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_datasource.o
../src/mapnik_datasource.cpp:150:63: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                              ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_featureset.o
../src/mapnik_featureset.cpp:101:63: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    return scope.Escape(Nan::New(constructor)->GetFunction()->NewInstance(1, &ext));
                                                              ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
1 warning generated.
  CXX(target) Release/obj.target/mapnik/src/mapnik_expression.o
  CXX(target) Release/obj.target/mapnik/src/mapnik_cairo_surface.o
  CXX(target) Release/obj.target/mapnik/src/mapnik_vector_tile.o
../src/mapnik_vector_tile.cpp:1399:74: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
    v8::Local<v8::Object> vt_obj = Nan::New(constructor)->GetFunction()->NewInstance(1, &ext);
                                                                         ^
/Users/mxn/.node-gyp/8.1.4/include/node/v8.h:3670:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
../src/mapnik_vector_tile.cpp:5232:28: warning: unused variable 'c' [-Wunused-variable]
            CairoSurface * c = mapnik::util::get<CairoSurface *>(closure->surface);
                           ^
2 warnings generated.
  SOLINK_MODULE(target) /path/to/node_modules/mapnik/lib/binding/mapnik.node
ld: warning: object file (/usr/local/lib/libmapnik-wkt.a(mapnik_wkt_generator_grammar.o)) was built for newer OSX version (10.11) than being linked (10.8)
ld: warning: object file (/usr/local/lib/libmapnik-json.a(mapnik_json_feature_grammar.o)) was built for newer OSX version (10.11) than being linked (10.8)
ld: warning: object file (/usr/local/lib/libmapnik-json.a(mapnik_json_generator_grammar.o)) was built for newer OSX version (10.11) than being linked (10.8)
ld: warning: object file (/usr/local/lib/libmapnik-json.a(mapnik_geometry_to_geojson.o)) was built for newer OSX version (10.11) than being linked (10.8)
ld: warning: object file (/usr/local/lib/libmapnik-json.a(generic_json.o)) was built for newer OSX version (10.11) than being linked (10.8)
ld: warning: object file (/usr/local/lib/libmapnik-json.a(mapnik_json_geometry_grammar.o)) was built for newer OSX version (10.11) than being linked (10.8)
ld: warning: object file (/usr/local/lib/libmapnik-json.a(mapnik_json_positions_grammar.o)) was built for newer OSX version (10.11) than being linked (10.8)
  COPY Release/mapnik.node
make: *** No rule to make target `../scripts/postinstall.sh', needed by `lib/binding/mapnik'.  Stop.
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
gyp ERR! System Darwin 16.6.0
gyp ERR! command "/usr/local/Cellar/node/8.1.4/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/path/to/node_modules/mapnik/lib/binding/mapnik.node" "--module_name=mapnik" "--module_path=/path/to/node_modules/mapnik/lib/binding"
gyp ERR! cwd /path/to/node_modules/mapnik
gyp ERR! node -v v8.1.4
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute 'node-gyp build --fallback-to-build --module=/path/to/node_modules/mapnik/lib/binding/mapnik.node --module_name=mapnik --module_path=/path/to/node_modules/mapnik/lib/binding' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/path/to/node_modules/mapnik/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:125:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:897:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)
node-pre-gyp ERR! System Darwin 16.6.0
node-pre-gyp ERR! command "/usr/local/Cellar/node/8.1.4/bin/node" "/path/to/node_modules/mapnik/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /path/to/node_modules/mapnik
node-pre-gyp ERR! node -v v8.1.4
node-pre-gyp ERR! node-pre-gyp -v v0.6.34
node-pre-gyp ERR! not ok 
Failed to execute 'node-gyp build --fallback-to-build --module=/path/to/node_modules/mapnik/lib/binding/mapnik.node --module_name=mapnik --module_path=/path/to/node_modules/mapnik/lib/binding' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mapnik@3.6.0 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mapnik@3.6.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mxn/.npm/_logs/2017-07-14T19_10_47_135Z-debug.log
```
</details>

**Comment by Algunenano** [comment]
The issue appears when installing the module with npm from source, i.e. using something like this:
`npm install --python=python2 --verbose --build-from-source=mapnik`

It seems this is due to the scripts folder being included in `.npmignore` so scrips/postinstall.sh  cannot be found. This commit by ifself should fix it: https://github.com/Algunenano/node-mapnik/commit/ec4cc8a07eb825bd6bf5d86e2278ac128b511e2d Let me know if you want a PR or feel free to cherry-pick it.

**Comment by springmeyer** [comment]
Thanks for figuring this one out @Algunenano - fixed in afe7c88. Will be fixed live in the node v3.6.3 release.
